manim/mobject/graphing/probability.py

```diff
@@ -339,27 +352,48 @@ def _add_x_axis_labels(self):
         self.x_axis.labels = labels
         self.x_axis.add(labels)
 
-    def _add_bars(self):
+    def _create_bar(self, bar_number: int, value: float) -> Rectangle:
+        """Creates a positioned bar on the chart.
+
+        Parameters
+        ----------
+        bar_number
+            Determines the x-position of the bar.
+        value
+            The value that determines the height of the bar.
+
+        Returns
+        -------
+        Rectangle
+            A positioned rectangle representing a bar on the chart.
+        """
+
+        # bar measurements relative to the axis
+
+        # distance from between the y-axis and the top of the bar
+        bar_h = abs(self.c2p(0, value)[1] - self.c2p(0, 0)[1])
+        # width of the bar
+        bar_w = self.c2p(self.bar_width, 0)[0] - self.c2p(0, 0)[0]
+
+        bar = Rectangle(
+            height=bar_h,
+            width=bar_w,
+            stroke_width=self.bar_stroke_width,
+            fill_opacity=self.bar_fill_opacity,
+        )
+
+        pos = UP if (value >= 0) else DOWN
+        bar.next_to(self.c2p(bar_number + 0.5, 0), pos, buff=0)
+        return bar
+
+    def _add_bars(self) -> None:
         self.bars = VGroup()
 
         for i, value in enumerate(self.values):
-            bar_h = abs(self.c2p(0, value)[1] - self.c2p(0, 0)[1])
-            bar_w = self.c2p(self.bar_width, 0)[0] - self.c2p(0, 0)[0]
-            bar = Rectangle(
-                height=bar_h,
-                width=bar_w,
-                stroke_width=self.bar_stroke_width,
-                fill_opacity=self.bar_fill_opacity,
-            )
-
-            pos = UP if (value >= 0) else DOWN
-            bar.next_to(self.c2p(i + 0.5, 0), pos, buff=0)
-            self.bars.add(bar)
-        if isinstance(self.bar_colors, str):
-            self.bars.set_color_by_gradient(self.bar_colors)
-        else:
-            self.bars.set_color_by_gradient(*self.bar_colors)
+            temp_bar = self._create_bar(bar_number=i, value=value)
+            self.bars.add(temp_bar)
 
+        self._update_colors()
         self.add_to_back(self.bars)
 
     def get_bar_labels(
```

Comment:
Well, it isn't really changed by this PR but the typing for this constructor is wrong for `label_constructor` it should be `Type[VMobject]` instead of `VMobject` see https://docs.python.org/3/library/typing.html#typing.Type